.github/workflows/build.yaml

```diff
@@ -14,10 +14,43 @@ concurrency:
 
 jobs:
   build-check:
-    name: "Check whether NTF builds correctly."
+    name: "Check whether NTF builds and passes unit tests"
     runs-on: ubuntu-latest
     timeout-minutes: 60
     steps:
       - uses: actions/checkout@v3
-      - name: Docker Cluster Workflow
-        run: docker build --tag ntf-core --platform linux/amd64 .
+
+      - name: Get cache name
+        id: get-hash
+        run: echo "ntf_deps_hash=`cat docker/build_deps.sh docker/build_test_ntf.sh docker/Dockerfile | shasum`" >> $GITHUB_OUTPUT
```

Comment:
If either of these files changes, it will cause cache regeneration. Note that it's impossible to include GH workflow files in this expression, because these files are not accessible for runner